optimum/onnxruntime/base.py

```diff
@@ -145,363 +146,13 @@ def __init__(
         if self.parent_model.use_cache is True and len(self.key_value_output_names) == 0:
             raise RuntimeError("Could not find the past key values in the provided model.")
 
-        if len(self.key_value_input_names) > 0:
-            self.use_past = True
-        else:
-            self.use_past = False
-
+        self.use_past = len(self.key_value_input_names) > 0
         self.use_fp16 = False
         for inp in session.get_inputs():
             if "past_key_values" in inp.name and inp.type == "tensor(float16)":
                 self.use_fp16 = True
                 break
 
-        if len(self.key_value_output_names) != 0:
-            # Attributes useful when computing the past key/values output shapes.
-            self.expected_key_symbolic_shape = None
-            self.expected_value_symbolic_shape = None
-            for output in self.session.get_outputs():
-                # To handle the case of multi-query attn where key and value are concatenated
-                if ".key_value" in output.name:
-                    expected_key_value_symbolic_shape = output.shape
-                    self.expected_key_symbolic_shape = (
-                        self.expected_value_symbolic_shape
-                    ) = expected_key_value_symbolic_shape[:-1] + [
-                        expected_key_value_symbolic_shape[-1] // 2,
-                    ]
-                elif ".key" in output.name:
-                    self.expected_key_symbolic_shape = output.shape
-                elif ".value" in output.name:
-                    self.expected_value_symbolic_shape = output.shape
-                # To handle the old case when past_key_values were following the format: past_key_values_{idx}
-                elif "key_values" in output.name:
-                    if self.expected_key_symbolic_shape is None:
-                        self.expected_key_symbolic_shape = output.shape
-                    else:
-                        self.expected_value_symbolic_shape = output.shape
-                if self.expected_key_symbolic_shape is not None and self.expected_value_symbolic_shape is not None:
-                    break
-
-            self.key_sequence_length_idx = -2
-            if (
-                isinstance(self.expected_key_symbolic_shape[-1], str)
-                and "sequence_length" in self.expected_key_symbolic_shape[-1]
-            ):
-                self.key_sequence_length_idx = -1
-
-            self.value_sequence_length_idx = -2
-            if (
-                isinstance(self.expected_value_symbolic_shape[-1], str)
-                and "sequence_length" in self.expected_value_symbolic_shape[-1]
-            ):
-                self.value_sequence_length_idx = -1
```

Comment:
this is not useful for seq2seq models?

Reply:
it's only used for computing the pkv shapes for decoder only models : https://github.com/huggingface/optimum/blob/v1.13.2/optimum/onnxruntime/base.py#L299 (not used by encoder/decoder models https://github.com/huggingface/optimum/blob/v1.13.2/optimum/onnxruntime/base.py#L504)